### PR TITLE
Add method operator (obj->method())

### LINKED ISSUE #150: method operator '->' is missing

### PR DESCRIPTION
**Pull request by tyru**
Fix #150

### CODE REVIEW

**Comment by tyru** [comment]
@mattn @blueyed can you review this?

File: autoload/vimlparser.vim [hunk 1]
```diff
@@ -3950,6 +3955,17 @@ function! s:ExprParser.parse_expr8() abort
...
       endif
       unlet node
+    elseif token.type ==# s:TOKEN_ARROW
+      let node = s:Node(s:NODE_METHOD)
+      let node.pos = token.pos
+      let node.left = left
+      let node.right = self.parse_expr8()
+      let node.lambda_rlist = s:NIL
+      if node.right.type !=# s:NODE_CALL
+        throw s:Err('Invalid method syntax', node.right.pos)
```
Comment (blueyed) [inline autoload/vimlparser.vim]: Can this error have more info?  Or is it enough?

File: test/test_method.vim [hunk 2]
```diff
@@ -0,0 +1,5 @@
+echo 'alice'->s:hello()
+echo 'alice'->s:hello('bob')
+echo [1,2,3]->map({i,n -> n * 2})
+echo 'john'->{s -> 'hello ' .. s }()
+echo 'john'->{s,b -> 'hello ' .. s .. ' ... and goodbye ' .. b .. '!' }('bob')
```
Comment (mattn) [inline test/test_method.vim]: please add one more test using `...` for argument.

Comment (tyru) [reply test/test_method.vim]: fixed e6c1c4c5ee162f1ca28c4041b5193b622428a3e8

File: autoload/vimlparser.vim (hunk 1 above)
Comment (tyru) [reply autoload/vimlparser.vim]: fixed 14221a4e2b21c66f7dd4550be8934cea3f7d3f6c

**Comment by tyru** [comment]
@mattn @blueyed Fixed. please review again.

**Comment by tyru** [comment]
@mattn @blueyed Thanks for review. It's time to merge 🎉

**Pull request merged**